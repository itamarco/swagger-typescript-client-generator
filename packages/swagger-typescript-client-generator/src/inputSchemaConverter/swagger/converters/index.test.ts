import * as Swagger from "swagger-schema-official"
import { convertSwaggerSchemaToNormalizedSchema } from "./index"

const testCases: Swagger.Schema[] = [
  {
    type: undefined,
    allOf: [
      {
        type: "boolean"
      },
      {
        type: "number"
      }
    ],
    description: "allOf should create intersection"
  }
]

test.each(testCases.map(item => [item.description, item]))(
  `converter converts scenario "%s"`,
  (description, definition) => {
    const spec = convertSwaggerSchemaToNormalizedSchema(definition as unknown)
    expect(spec).toMatchSnapshot()
  }
)
