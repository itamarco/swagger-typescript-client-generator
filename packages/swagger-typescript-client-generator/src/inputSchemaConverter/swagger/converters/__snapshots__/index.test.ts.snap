// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converter converts scenario "allOf should create intersection" 1`] = `
Object {
  "items": Array [
    Object {
      "default": undefined,
      "description": undefined,
      "enum": undefined,
      "type": "boolean",
    },
    Object {
      "default": undefined,
      "description": undefined,
      "enum": undefined,
      "format": undefined,
      "maximum": undefined,
      "minimum": undefined,
      "type": "number",
    },
  ],
  "type": "intersection",
}
`;
